File: app/src/main/java/com/hex/evegate/api/dto/NowPlayingDto.java
package com.hex.evegate.api.dto;

import com.google.gson.annotations.SerializedName;

public class NowPlayingDto {

    @SerializedName("cache") private String cache;

    @SerializedName("listeners") private Listeners listeners;

    @SerializedName("now_playing") private NowPlaying now_playing;

    @SerializedName("song_history") private SongHistory[] song_history;

    @SerializedName("station") private Station station;

    @SerializedName("live") private Live live;

    @SerializedName("playing_next") private PlayingNext playing_next;

    public String getCache ()
    {
        return cache;
    }

    public void setCache (String cache)
    {
        this.cache = cache;
    }

    public Listeners getListeners ()
    {
        return listeners;
    }

    public void setListeners (Listeners listeners)
    {
        this.listeners = listeners;
    }

    public NowPlaying getNow_playing ()
    {
        return now_playing;
    }

    public void setNow_playing (NowPlaying now_playing)
    {
        this.now_playing = now_playing;
    }

    public SongHistory[] getSong_history ()
    {
        return song_history;
    }

    public void setSong_history (SongHistory[] song_history)
    {
        this.song_history = song_history;
    }

    public Station getStation ()
    {
        return station;
    }

    public void setStation (Station station)
    {
        this.station = station;
    }

    public Live getLive ()
    {
        return live;
    }

    public void setLive (Live live)
    {
        this.live = live;
    }

    public PlayingNext getPlaying_next ()
    {
        return playing_next;
    }

    public void setPlaying_next (PlayingNext playing_next)
    {
        this.playing_next = playing_next;
    }

    @Override
    public String toString()
    {
        return "ClassPojo [cache = "+cache+", listeners = "+listeners+", now_playing = "+now_playing+", song_history = "+song_history+", station = "+station+", live = "+live+", playing_next = "+playing_next+"]";
    }
}

File: app/src/main/java/com/hex/evegate/api/dto/NowPlayingDto.kt
package com.hex.evegate.api.dto

import com.google.gson.annotations.SerializedName

data class NowPlayingDto (
    @SerializedName("cache") var cache: String,
    @SerializedName("listeners") var listeners: Listeners,
    @SerializedName("now_playing") var now_playing: NowPlaying,
    @SerializedName("song_history") var song_history: Array<SongHistory>,
    @SerializedName("station") var station: Station,
    @SerializedName("live") var live: Live,
    @SerializedName("playing_next") var playing_next: PlayingNext
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as NowPlayingDto

        if (cache != other.cache) return false
        if (listeners != other.listeners) return false
        if (now_playing != other.now_playing) return false
        if (!song_history.contentEquals(other.song_history)) return false
        if (station != other.station) return false
        if (live != other.live) return false
        if (playing_next != other.playing_next) return false

        return true
    }

    override fun hashCode(): Int {
        var result = cache.hashCode()
        result = 31 * result + listeners.hashCode()
        result = 31 * result + now_playing.hashCode()
        result = 31 * result + song_history.contentHashCode()
        result = 31 * result + station.hashCode()
        result = 31 * result + live.hashCode()
        result = 31 * result + playing_next.hashCode()
        return result
    }
}
